sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala

```diff
@@ -92,6 +92,19 @@ private[sql] object QueryCompilationErrors extends QueryErrorsBase {
         pivotVal.toString, pivotVal.dataType.simpleString, pivotCol.dataType.catalogString))
   }
 
+  def unpivotRequiresValueColumns(ids: Seq[NamedExpression]): Throwable = {
+    new AnalysisException(
+      errorClass = "UNPIVOT_REQUIRES_VALUE_COLUMNS",
+      messageParameters = Array(ids.map(id => toSQLId(id.toString)).mkString(", ")))
+  }
+
+  def unpivotValDataTypeMismatchError(values: Seq[NamedExpression]): Throwable = {
+    val dataTypes = values.map(_.dataType).toSet.map((dt: DataType) => toSQLType(dt))
```

Comment:
shall we keep the order? `values.map(v => toSQLType(v.dataType))`

Reply:
`toSet` removes duplicated types, but I think it's easier for users to find out which value column is problematic, if they can match type and columns by ordinal.

Reply:
Wide tables would list a lot of types, possibly the same types multiple times, which makes it harder to spot the incompatible types. And after the tenth ordinal, the error message gets messy and less helpful.

Alternatively, the error could list a few (3) columns per type. That would make it easier to identify the one / few column(s) that are incompatible with all others.

Reply:
Error with all columns' data types in column order would look like this:
```
Unpivot value columns must share a least common type, some types do not: ["STRING", "INT", "INT", "INT", "INT", "BIGINT", "BIGINT"]
```

Error with all types and at most three columns would look like this:
```
Unpivot value columns must share a least common type, some types do not: ["STRING" (`str1#6`), "INT" (`int1#8`, `int2#190`, `int3#191`, …), "BIGINT" (`long1#9L`, `long2#193L`)]
```

Reply:
Added three columns per datatype to the error: ad8b5d3ccfb7b31a0577fe2fe042d16ee14e9c79